src/@itwinjs-sandbox/components/imodel-selector/IModelSelector.tsx

```diff
@@ -0,0 +1,41 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import React, { FunctionComponent } from "react";
+import { Select } from "@bentley/ui-core";
+import "@bentley/icons-generic-webfont/dist/bentley-icons-generic-webfont.css";
+import { SampleIModels } from "@itwinjs-sandbox/SampleIModels";
+
+// The Props and State for this sample component
+interface IModelSelectorProps {
+  iModelNames: SampleIModels[];
+  iModelName: SampleIModels;
+  onIModelChange: (iModelName: SampleIModels) => void;
```

Comment:
should this be async?

Reply:
Not necessarily. At the end of the day, this component is just a select component. It would probably be better served as a regular function for reusability's sake.